src/components/CandidateAvatar/CandidateAvatar.tsx

```diff
@@ -1,21 +1,31 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import bem from '../../utils/bem';
 import styles from './CandidateAvatar.scss';
 
+interface Props {
+    /** Path to the avatar image resource */
+    imageUrl: string;
+    /** The match percentage; ranging from 0 (bad) - 100 (good) */
+    matchPercentage: number;
+    /** Whether to show the match percentage when hovering the image */
+    showPercentageOnHover: boolean;
+    /** Avatar size (1:1 aspect ratio); should be even number */
+    size: number;
+};
+
 const { block, elem } = bem('CandidateAvatar', styles);
 
-const context = percentage => {
-    if (percentage <= 33) {
-        return 'bad';
-    }
-    if (percentage <= 66) {
-        return 'warning';
+const CandidateAvatar = (props: Props) => {
+    function getContext(percentage: number): string {
```

Comment:
Here, return type can be narrowed down from `string` to `'bad' | 'warning' | 'good'`